template.go

```diff
@@ -160,6 +170,17 @@ func readYAMLRequest(base string, contents []byte, templateArgs map[string]strin
 		opts.ROpts.ThriftFile = thriftFileURL.Path
 	}
 
+	switch strings.ToLower(t.Encoding) {
+	case "thrift":
+		opts.ROpts.Encoding = encoding.Thrift
+	case "json":
+		opts.ROpts.Encoding = encoding.JSON
+	case "protobuf":
+		opts.ROpts.Encoding = encoding.Protobuf
```

Comment:
When using the option -e, --encoding, encoding.Protobuf will match with `proto` and not `protobuf`.